e3nn/tensor_product.py

```diff
@@ -73,39 +73,70 @@ def forward(self, features_1, features_2):
 
 
 class LearnableTensorProduct(torch.nn.Module):
-    def __init__(self, Rs_mid_1, Rs_mid_2, mul_mid, Rs_out, selection_rule=o3.selection_rule):
+    def __init__(self, Rs_mid_1, Rs_mid_2, Rs_out,
+                 selection_rule=o3.selection_rule, groups=1):
         super().__init__()
-        self.mul_mid = mul_mid
+        self.groups = groups
         self.tp = TensorProduct(Rs_mid_1, Rs_mid_2, selection_rule)
-        self.lin = Linear(mul_mid * self.tp.Rs_out, Rs_out)
+        self.lin = Linear(groups * self.tp.Rs_out, Rs_out)
 
     def forward(self, x1, x2):
         """
         :return:         tensor [..., channel]
         """
         # split into mul x Rs
         *size, _ = x1.shape
-        x1 = x1.view(*size, self.mul_mid, -1)
-        x2 = x2.view(*size, self.mul_mid, -1)
+        x1 = x1.view(*size, self.groups, -1)
+        x2 = x2.view(*size, self.groups, -1)
 
         x = self.tp(x1, x2).view(*size, -1)
         x = self.lin(x)
         return x
 
 
 class LearnableBispectrum(torch.nn.Module):
```

Comment:
Just wondering if we can find a more "down to earth" that says just what it does and not how it comes from because I have the impression that at the end it's simpler that where it comes from